api/clients/eigenda_client.go

```diff
@@ -178,18 +199,34 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 
 	alreadyWaitingForDispersal := false
 	alreadyWaitingForFinalization := false
+	var latestBlobStatus grpcdisperser.BlobStatus
 	for {
 		select {
 		case <-ctx.Done():
-			errChan <- fmt.Errorf("timed out waiting for EigenDA blob to confirm blob with request id=%s: %w", base64RequestID, ctx.Err())
+			// We can only land here if blob is still in
+			// 1. processing or dispersing status: waiting to land onchain
+			// 2. or confirmed status: landed onchain, waiting for finalization
+			// because all other statuses return immediately below.
+			//
+			// Assuming that the timeout is correctly set (long enough to both land onchain + finalize),
+			// 1. means that there is a problem with EigenDA, so we return 503 to let the batcher failover to ethda
+			// 2. means that there is a problem with Ethereum, so we return 500.
+			//    batcher would most likely resubmit another blob, which is not ideal but there isn't much to be done...
+			//    eigenDA v2 will have idempotency so one can just resubmit the same blob safely.
```

Comment:
Not sure I'm following your comment. v2 will have idempotency. You can resubmit the same blob, and disperser will just give you the blobStatus blob right away. Same thing as sending a PUT request with idempotency key. If its already there, it doesn't get written twice. That's the whole point of idempotency.

Reply:
Ok, it is one way to achieve idempotency within some time window. It is interesting that, we can achieve it by making it impossible to disperser to accept blobs for that time window. 

However, once that time window passed. User can disperse the same blob twice.

Reply:
Ah I see. That is true, depending on whether we keep the metadata table (history of requests) forever or not.
I had another comment somewhere in this PR I think mentioning this regarding FAILED status. did you answer it?

Reply:
I think how it is going to be implemented with bins or leaky bucket alike, like how current rate limitor works. The bin window is about 10-30min? (not super sure). But in the long term, it is going to be an onchain parameter I think

Reply:
Oh? I think I misunderstood your previous comment then.
That seems wrong then? Idempotency shouldn't have anything to do with rate limits and small window size... it should have to do with whether we already received the blob. So we should just check if that blobHeader is already in the dynamodb metadata table (or wherever we store those things in v2), and if yes, then return the blobStatus for that blob instead of redispersing it. No?